pandas/core/aggregation.py

```diff
@@ -451,37 +463,49 @@ def transform(
     return result
 
 
-def transform_dict_like(obj, func, *args, **kwargs):
+def transform_dict_like(
+    obj: FrameOrSeries,
+    func: Dict[Label, Union[AggFuncTypeBase, List[AggFuncTypeBase]]],
+    *args,
+    **kwargs,
+):
     """
     Compute transform in the case of a dict-like func
     """
     from pandas.core.reshape.concat import concat
 
+    if len(func) == 0:
+        raise ValueError("no results")
```

Comment:
where is this raised to? is it user facing?

Reply:
Yes - this is raised directly to user. The error type and message agrees with 1.1.x. Can change/improve if desired.

Reply:
ok i think this needs to be more explict on what is happening, e.g. an empty function specification was provided. otherwise lgtm.